iroh-resolver/src/balanced_tree.rs

```diff
@@ -77,11 +77,18 @@ fn stream_balanced_tree(
         let mut tree: VecDeque<Vec<(Cid, LinkInfo)>> = VecDeque::new();
         tree.push_back(Vec::with_capacity(degree));
 
+        let hash_par: usize = 8;
+
+        let in_stream = in_stream.err_into::<anyhow::Error>().map(|chunk| {
+            tokio::task::spawn_blocking(|| {
+                chunk.and_then(|chunk| TreeNode::Leaf(chunk.freeze()).encode())
+            }).err_into::<anyhow::Error>()
+        }).buffered(hash_par).map(|x| x.and_then(|x| x));
```

Comment:
Not sure if there is a nicer way to do this than `.map(|x| x.and_then(|x| x))`.

Reply:
can't you use flatten?

Reply:
Now that you mention it... But apparently not.

<img width="781" alt="image" src="https://user-images.githubusercontent.com/248257/197764056-f17bfdf0-45e8-4257-a0d3-7c9bc150f4bb.png">